prefect_aws/workers/glue_job_worker.py

```diff
@@ -0,0 +1,156 @@
+import logging
+import time
+from typing import Any, Optional
+
+import anyio
+from prefect.server.schemas.core import FlowRun
+from prefect.utilities.asyncutils import run_sync_in_worker_thread
+from prefect.workers.base import (
+    BaseJobConfiguration,
+    BaseVariables,
+    BaseWorker,
+    BaseWorkerResult,
+)
+from pydantic import Field
+
+from prefect_aws import AwsCredentials
+
+_GlueJobClient = Any
+
+
+class GlueJobWorkerConfiguration(BaseJobConfiguration):
+    """
+    Job configuration for a Glue Job.
+    """
+
+    job_name: str = Field(
+        ...,
+        title="AWS Glue Job Name",
+        description="The name of the job definition to use.",
+    )
+    arguments: Optional[dict] = Field(
+        default=None,
+        title="AWS Glue Job Arguments",
+        description="The job arguments associated with this run.",
+    )
+    job_watch_poll_interval: float = Field(
+        default=60.0,
+        description=(
+            "The amount of time to wait between AWS API calls while monitoring the "
+            "state of an Glue Job."
+        ),
+    )
+    aws_credentials: Optional[AwsCredentials] = Field(default_factory=AwsCredentials)
```

Comment:
Since this is has a `default_factory`, we can remove the `Optional` type hint since we don't want `aws_credentials` ever to be `None`.
```suggestion
    aws_credentials: AwsCredentials = Field(default_factory=AwsCredentials)
```

Reply:
Thanks, fix it
https://github.com/PrefectHQ/prefect-aws/pull/308/commits/71934c2101735a1b0b6dfce81a45448f8f7b6722